cloudinit/net/__init__.py

```diff
@@ -266,28 +262,10 @@ def is_vlan(devname):
     return 'DEVTYPE=vlan' in uevent.splitlines()
 
 
-def is_connected(devname):
-    # is_connected isn't really as simple as that.  2 is
-    # 'physically connected'. 3 is 'not connected'. but a wlan interface will
-    # always show 3.
-    iflink = read_sys_net_safe(devname, "iflink")
-    if iflink == "2":
-        return True
-    if not is_wireless(devname):
-        return False
-    LOG.debug("'%s' is wireless, basing 'connected' on carrier", devname)
-    return read_sys_net_safe(devname, "carrier",
-                             translate={'0': False, '1': True})
-
-
```

Comment:
It would be good to refactor this to match what's in cloudinit/net/__init__.py:find_fallback_nic_linux() where we're looking for connected devices...

```
        carrier = read_sys_net_int(interface, 'carrier')                       
        if carrier:                                                            
            connected.append(interface)                                        
            continue                                                           
        # check if nic is dormant or down, as this may make a nick appear to   
        # not have a carrier even though it could acquire one when brought     
        # online by dhclient                                                   
        dormant = read_sys_net_int(interface, 'dormant')                       
        if dormant:                                                            
            possibly_connected.append(interface)                               
            continue                                                           
        operstate = read_sys_net_safe(interface, 'operstate')                  
        if operstate in ['dormant', 'down', 'lowerlayerdown', 'unknown']:      
            possibly_connected.append(interface)                               
            continue                                             
```
We could keep your drop but replace it with this implementation, and then refactor the fallback code to use this code.

Reply:
I'm not 100% sure I follow, so let me echo back what I think you're proposing: we could improve the implementation of `find_fallback_nic_on_linux` by refactoring the "connected status" code it currently uses out to a separate function/method (which would have a similar purpose to the one that's being dropped)?

If so, would a reasonable path forward be to add a comment to https://bugs.launchpad.net/cloud-init/+bug/1884606 so that the person who picks up that refactor takes this on-board at that point?

Reply:
@OddBlock 

That sounds fine.  I'll make a comment there.

Reply:
Thanks!